ProtocolHandler.pm

```diff
@@ -36,7 +52,165 @@ sub new {
 	
 	${*$sock}{contentType} = $mime;
 
-	return $sock;
+    if (defined($sock)) {
+		${*$sock}{'vars'} = {                   # variables which hold state for this instance:
```

Comment:
Do these values have to live in the socket object? Wouldn't $song->pluginData() work here, too?

Reply:
That's the way it works in plugins philippe showed to me. I think is net , concise and - best of all -  concluded in just one class,  but maybe is possible also in your way, don't know.